pytorch_lightning/trainer/distrib_parts.py

```diff
@@ -559,7 +559,7 @@ def check_gpus_data_type(gpus):
     :return: return unmodified gpus variable
     """
 
-    if gpus is not None and type(gpus) not in (int, str, list):
+    if gpus is not None and not isinstance(gpus, (int, str, list)):
```

Comment:
I think the tests are failing because bool is a subclass of int, therefore the instance check will include bool. `gpus=False` is a valid trainer input. I think that's why we had to keep the check with `type` there rather than `isinstance`.